devicetypes/smartthings/zigbee-dimmer-with-motion-sensor.src/zigbee-dimmer-with-motion-sensor.groovy

```diff
@@ -0,0 +1,162 @@
+					log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+					sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+				} else {
+					log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+				}
+			} else if (device.getDataValue("manufacturer") == "sengled" && descMap && descMap.clusterInt == 0x0008 && descMap.attrInt == 0x0000) {
+				// This is being done because the sengled element touch/classic incorrectly uses the value 0xFF for the max level.
+				// Per the ZCL spec for the UINT8 data type 0xFF is an invalid value, and 0xFE should be the max.  Here we
+				// manually handle the invalid attribute value since it will be ignored by getEvent as an invalid value.
+				// We also set the level of the bulb to 0xFE so future level reports will be 0xFE until it is changed by
+				// something else.
+				if (descMap.value.toUpperCase() == "FF") {
+					descMap.value = "FE"
+				}
+				sendHubCommand(zigbee.command(zigbee.LEVEL_CONTROL_CLUSTER, 0x00, "FE0000").collect { new physicalgraph.device.HubAction(it) }, 0)
+				map = zigbee.getEventFromAttrData(descMap.clusterInt, descMap.attrInt, descMap.encoding, descMap.value)
+			} else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+				def zs = new ZoneStatus(zigbee.convertToInt(descMap.value, 16))
+				map = translateZoneStatus(zs)
+			} else if (descMap?.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS && descMap?.value) {
+				map = translateZoneStatus(new ZoneStatus(zigbee.convertToInt(descMap?.value)))
+			} else {
+				log.warn "DID NOT PARSE MESSAGE for description : $description"
+				log.debug "${descMap}"
+			}
+		}
+	} else if (map.name == "level" && map.value == 0) {
+		map = [:]
+	}
+
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+
+	return result
+}
+
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+
+	return translateZoneStatus(zs)
+}
+
+private Map translateZoneStatus(ZoneStatus zs) {
+	// Some sensor models that use this DTH use alarm1 and some use alarm2 to signify motion
+	return getMotionResult(zs.isAlarm1Set() || zs.isAlarm2Set())
+}
+
+private Map getMotionResult(value) {
+	def descriptionText = value ? "${device.displayName} detected motion" : "${device.displayName} motion has stopped"
+	return [
+			name			: 'motion',
+			value			: value ? 'active' : 'inactive',
+			descriptionText : descriptionText,
+			translatable	: true
+	]
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value) {
+	zigbee.setLevel(value)
+}
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	return zigbee.onOffRefresh() + zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+}
+
+def refresh() {
+	state.enrollResponseCount = 0
```

Comment:
```suggestion
```

Reply:
d'oh! And wow! That's a cool new feature.

Reply:
Yeah, and it looks like you would have been able to directly apply it through github (although i don't know what that would look like).  But apparently we were in a hurry so I just did what I knew.

Reply:
That's fine with me! There's a box below the suggestion, which is now greyed out, that says "Apply suggestion". That's pretty sweet.